### PR TITLE
Generate azure subs config file

### PR DESCRIPTION
**Pull request by tbarlow12**

### CODE REVIEW

File: tools/c7n_org/azuresubs.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+        output = '../azure_config.yaml'
+
+        with open(output, 'w') as f:
+            f.write(yaml.safe_dump({
+                'accounts': results
+            }, default_flow_style=False))
+        '''print(
+            yaml.safe_dump(
+                {'accounts': results},
```
Comment (kapilt) [inline tools/c7n_org/azuresubs.py]: per discussion at sprint, let's stick this in 'subscriptions' and switch to `subscription_id`

File: tools/c7n_org/azuresubs.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+        results = []
+        for sub in subs:
+            sub_info = {
+                'subscription_id': sub['subscriptionId'],
+                'name': sub['displayName']
+            }
+            results.append(sub_info)
+
+        output = '../azure_config.yaml'
```
Comment (kapilt) [inline tools/c7n_org/azuresubs.py]: this should be a cli parameter instead fo hardcode, and generally a hardcode shouldn't reach outside the current directory

File: tools/c7n_org/azuresubs.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    """
+
+    tenant_auth_variables = [
+        'AZURE_TENANT_ID', 'AZURE_SUBSCRIPTION_ID',
+        'AZURE_CLIENT_ID', 'AZURE_CLIENT_SECRET'
+    ]
+
+    # Set credentials with environment variables if all
+    # required variables are present
+    if all(k in os.environ for k in tenant_auth_variables):
```
Comment (kapilt) [inline tools/c7n_org/azuresubs.py]: at the moment the entire function body is behind this if condition, i'd suggest inverting the condition (ie `if not`), giving the user a warning message about the things your looking for and returning. Then the rest of the function body can be dedented.

File: tools/c7n_org/azuresubs.py (hunk 2 above)
Comment (tbarlow12) [reply tools/c7n_org/azuresubs.py]: totally forgot I left this in here. thanks

**Comment by kapilt** [comment]
sounds good, we should also support service account and user creds for the subscription discovery.

**Pull request closed without merging**